src/terminal/formatter.zig

```diff
@@ -1269,6 +1354,26 @@ pub const PageFormatter = struct {
         }
     }
 
+    /// Write a string with HTML escaping. Used for escaping href attributes
+    /// and other HTML attribute values.
+    fn writeHtmlEscaped(
```

Comment:
We should already have this helper somewhere, `writeCodepoint` I think. Should be reused.